service-config/src/main/java/com/palantir/config/service/ServiceDiscoveryConfiguration.java

```diff
@@ -60,15 +61,18 @@
      */
     @Lazy
     @SuppressWarnings("checkstyle:designforextension")
+    @JsonIgnore
     public Map<String, ServiceConfiguration> getServices() {
         Map<String, ServiceConfiguration> intializedServices = new HashMap<String, ServiceConfiguration>();
 
         for (Map.Entry<String, ServiceConfiguration> entry : originalServices().entrySet()) {
-            ServiceConfiguration serviceConfig = entry.getValue();
-            Optional<BearerToken> apiToken = serviceConfig.apiToken().or(getDefaultApiToken());
-            Optional<SslConfiguration> security = serviceConfig.security().or(getDefaultSecurity());
-            intializedServices.put(entry.getKey(),
-                    ImmutableServiceConfiguration.of(apiToken, security, serviceConfig.uris()));
+            ServiceConfiguration originalConfig = entry.getValue();
+            Optional<BearerToken> apiToken = originalConfig.apiToken().or(defaultApiToken());
```

Comment:
I think this would be more clear as follows:
````
ServiceConfiguration configWithDefaults = ServiceConfiguration.builder()
    .apiToken(serviceConfig.apiToken().or(defaultApiToken())
    .security(serviceConfig.security().or(defaultSecurity())
    .uris(serviceConfig.uris())
    .build();
````